test/missing.jl

```diff
@@ -121,11 +128,21 @@ end
         @test ismissing(f(missing))
     end
 
-    @test zero(Union{Int, Missing}) === 0
-    @test zero(Union{Float64, Missing}) === 0.0
+    for T in (Int, Float64)
+        @test zero(Union{T, Missing}) === T(0)
+        @test one(Union{T, Missing}) === T(1)
+        @test oneunit(Union{T, Missing}) === T(1)
+    end
+
+    for T in (Unit,)
+        @test zero(Union{T, Missing}) === T(0)
+        @test one(Union{T, Missing}) === 1
+        @test oneunit(Union{T, Missing}) === T(1)
+    end
+
     @test_throws MethodError zero(Any)
-    @test_throws MethodError zero(String)
-    @test_throws MethodError zero(Union{String, Missing})
```

Comment:
This test would still be good to have AFAICT.

Reply:
Ok, will only apply to `zero`

Reply:
Doesn't it throw the same error for `one` and `oneunit`?

Reply:
`one(String) == ""` because `"foo" * "" == "foo"`

Reply:
Ah, I thought we had decided against it. Anyway, you can use `Symbol` instead of `String`.

Reply:
You'll also have to add a special-case for the new methods in the `ambiguous` test (which explains the failures).